subprojects/logging/src/main/java/org/gradle/internal/logging/LoggingCommandLineConverter.java

```diff
@@ -18,36 +18,28 @@
 
 import com.google.common.collect.BiMap;
 import com.google.common.collect.HashBiMap;
-import org.apache.commons.lang.StringUtils;
 import org.gradle.api.logging.LogLevel;
-import org.gradle.api.logging.configuration.ConsoleOutput;
 import org.gradle.api.logging.configuration.LoggingConfiguration;
 import org.gradle.api.logging.configuration.ShowStacktrace;
 import org.gradle.cli.AbstractCommandLineConverter;
 import org.gradle.cli.CommandLineArgumentException;
 import org.gradle.cli.CommandLineParser;
 import org.gradle.cli.ParsedCommandLine;
+import org.gradle.internal.buildoption.BuildOption;
 
 import java.util.Arrays;
 import java.util.HashSet;
-import java.util.Locale;
-import java.util.Map;
+import java.util.List;
 import java.util.Set;
 
 public class LoggingCommandLineConverter extends AbstractCommandLineConverter<LoggingConfiguration> {
-    public static final String DEBUG = "d";
-    public static final String DEBUG_LONG = "debug";
-    public static final String WARN = "w";
-    public static final String WARN_LONG = "warn";
-    public static final String INFO = "i";
-    public static final String INFO_LONG = "info";
-    public static final String QUIET = "q";
-    public static final String QUIET_LONG = "quiet";
-    public static final String CONSOLE = "console";
-    public static final String FULL_STACKTRACE = "S";
-    public static final String FULL_STACKTRACE_LONG = "full-stacktrace";
-    public static final String STACKTRACE = "s";
-    public static final String STACKTRACE_LONG = "stacktrace";
+    private final List<BuildOption<LoggingConfiguration>> buildOptions = new LoggingConfigurationBuildOptionFactory().create();
+    public static final String DEBUG = LoggingConfigurationBuildOptionFactory.DebugOption.SHORT_OPTION;
```

Comment:
DEBUG, WARN, INFO and QUIET are all command-line options that affect the log level. There's only one build option here - the log level - and it has several command-line options that affect it. They should all be merged into the LogLevel build option.

Reply:
Perhaps we ought to add `--log-level=FOO` as a single option? Of course, we must leave current mechanisms for now but perhaps we ought to deprecate them (in a separate change of course)